stdlib/os/__init__.pyi

```diff
@@ -624,8 +626,8 @@ if sys.platform != "win32":
     if sys.platform != "darwin":
         if sys.version_info >= (3, 10):
             RWF_APPEND: int  # docs say available on 3.7+, stubtest says otherwise
-        def preadv(__fd: int, __buffers: Iterable[bytes], __offset: int, __flags: int = ...) -> int: ...
-        def pwritev(__fd: int, __buffers: Iterable[bytes], __offset: int, __flags: int = ...) -> int: ...
+        def preadv(__fd: int, __buffers: Sequence[WriteableBuffer], __offset: int, __flags: int = ...) -> int: ...
```

Comment:
Looking at the C code, this only needs `__getitem__` and `__len__`, not an object that supports the sequence protocol. Can you use `_typeshed.SupportsLenAndGetItem`?

Reply:
https://github.com/python/cpython/blob/0319cd6825f3c247a875c91493a38992fb33a5b3/Modules/posixmodule.c#L9749

Doesn't this mean it needs to be a Sequence?

Reply:
No, PySequence_Check checks only for `__getitem__`. It is not equivalent to Sequence in Python.

Reply:
Okay. All four functions do the same check, so I've done that for all of them.

What about `ReadOnlyBuffer` vs `ReadableBuffer` in my comment above?

Reply:
Thanks. You're correct in your comment; the C code indicates they take any readable or writable buffer.

Reply:
Okay, done that as well.